packages/shared/src/hooks/utils/useGetShortUrl.ts

```diff
@@ -0,0 +1,45 @@
+import { useQueryClient } from '@tanstack/react-query';
+import request from 'graphql-request';
+import { useCallback } from 'react';
+import { graphqlUrl } from '../../lib/config';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { GET_SHORT_URL_QUERY } from '../../graphql/urlShortener';
+import { addTrackingQueryParams } from '../../lib/share';
+
+interface UseGetShortUrlResult {
+  getShortUrl: (url: string, cid?: string) => Promise<string>;
+}
+
+export const useGetShortUrl = (): UseGetShortUrlResult => {
+  const { user, isAuthReady } = useAuthContext();
+  const queryClient = useQueryClient();
+
+  const getShortUrl = useCallback(
+    async (url: string, cid?: string) => {
+      if (!url || !isAuthReady || !user) {
+        return url;
+      }
+
+      const trackingUrl = cid
+        ? addTrackingQueryParams({ link: url, userId: user?.id, cid })
```

Comment:
More of a question I see this function used here as well as on all components, wasn't there a better abstraction we could do?
Non blocking, and didn't really follow all steps right, so might be invalid.

Reply:
I kept 2 separate concerns:
1. Adding the query params to the URL
2. Sending the URL to a URL shortener

It's because sometimes we need to use the unminified URL for other purposes or when we can easily cache it. So the `addTrackingQueryParams` is definitely not used on all of the components. 

The `getShortUrl` function accepts 2 parameters, if you supply the CID as a second param, it will do both of these steps. I think I can rewrite more of the components to use this notation.

Reply:
I kept one place where we sitll need it because of the quirkiness of how we implemented the `SocialShare` component, that wraps the `SocialShareList`. Unless we want to refactor that as well 🤔